main.py

```diff
@@ -923,6 +924,13 @@ def ui_access_wrapper(self, *args, **kwargs):
             r'/initialize_android_test_data',
             android_e2e_config.InitializeAndroidTestDataHandler))
 
+# Redirect all routes handled using angular router to the oppia root page.
+for route in constants.ROUTES_REGISTERED_WITH_FRONTEND.values():
+    URLS.append(
+        get_redirect_route(
+            r'/%s' % route,
+            oppia_root.OppiaRootPage))
+
 # 404 error handler (Needs to be at the end of the URLS list).
 URLS.append(get_redirect_route(r'/<:.*>', base.Error404Handler))
```

Comment:
Do we still need for the 404 page?

Reply:
yeah, 404 is still handled through here and then redirected to oppia-root.mainpage.html